src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -125,6 +127,24 @@ this.FirefoxMonitor = {
         },
       });
     });
+
+    AddonManager.addAddonListener(this);
+  },
+
+  onUninstalling(addon) {
+    this.handleDisableOrUninstall(addon);
+  },
+
+  onDisabled(addon) {
+    this.handleDisableOrUninstall(addon);
+  },
+
+  handleDisableOrUninstall(addon) {
+    if (addon.id !== gExtension.id) {
+      return;
+    }
+    AddonManager.removeAddonListener(this);
+    addon.uninstall();
```

Comment:
This seems fine given our conversation on IRC, but it looks fishy enough that it would be great to have a comment about it.

Thanks!